packages/design-system/src/components/WIP/Accordion/Primitive/CollapsiblePanel.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { forwardRef, ReactChild, Ref, useState, useEffect, HTMLAttributes } from 'react';
+import classNames from 'classnames';
+import { unstable_useId as useId } from 'reakit';
+
+import { DataAttributes } from 'src/types';
+
+import CollapsiblePanelHeader from './CollapsiblePanelHeader';
+import { PanelHeaderAction } from './types';
+import styles from './CollapsiblePanel.module.scss';
+
+type CollapsiblePanelPropsType = {
+	children: ReactChild;
+	managed?: boolean;
+	expanded?: boolean;
+	index?: number;
+	title: string;
+	action?: PanelHeaderAction;
+	size?: 'S' | 'M';
+	metadata?: ReactChild[];
+	isFirst?: boolean;
+	isLast?: boolean;
+	disabled?: boolean;
+	onToggleExpanded?: (index: number) => void;
+} & Omit<HTMLAttributes<HTMLDivElement>, 'className' | 'style'> &
+	DataAttributes;
+
+const CollapsiblePanel = forwardRef(
+	(
+		{
+			id,
+			children,
+			managed,
+			expanded,
+			index,
+			onToggleExpanded,
+			title,
+			action,
+			size,
+			metadata,
+			isFirst = false,
+			isLast = false,
+			disabled = false,
+			...rest
+		}: CollapsiblePanelPropsType,
+		ref: Ref<HTMLButtonElement>,
+	) => {
+		const [localExpanded, setLocalExpanded] = useState(false);
+
+		const { id: reakitId } = useId();
+		const componentId = id || reakitId;
+		const controlId = `CollapsiblePanel__control--${componentId}`;
+		const sectionId = `CollapsiblePanel__content--${componentId}`;
+
+		useEffect(() => {
+			if (managed && expanded != localExpanded) {
+				setLocalExpanded(!!expanded);
+			}
+		}, [expanded, managed, localExpanded]);
+
+		function handleToggleExpanded() {
+			if (onToggleExpanded && managed && (index || index === 0)) {
+				onToggleExpanded(index);
+			} else {
+				setLocalExpanded(!localExpanded);
+			}
+		}
+
+		return (
+			<div
+				className={classNames(styles.panelWrapper, {
+					[styles.panelWrapper__alone]: !managed,
+					[styles.panelWrapper__first]: isFirst,
+					[styles.panelWrapper__last]: isLast,
+					[styles.panelWrapper__notLast]: managed && !isLast,
+				})}
+				{...rest}
+			>
+				<CollapsiblePanelHeader
+					controlId={controlId}
+					sectionId={sectionId}
+					expanded={disabled ? false : localExpanded}
```

Comment:
I agree that your proposal is shorter, but I find it less easy to read/explicit.
WDYT ?

Reply:
For me, this is about how it is fast to read it when you scan a code.

`disabled ? false` seems weird to me because `disabled` must be _true_ to set _false_ for `expanded`. It 's a little slower to read it than `!disabled` that means we set _false_ if expression is _false_. I don't have to think in "reverse way" to know what is expected behavior.

But it also depends how your brain is comfortable with this kind of write

Anyway it is just a suggestion, it's up to you